docs/cookbook/html-conversion.md

```diff
@@ -0,0 +1,980 @@
+
+<AposCodeBlock>
+::: v-pre
+```django
+{% extends "layout.html" %}
+
+{% import 'fragments/header.html' as header %}
+
+{% block beforeMain %}
+    {{ super() }}
+    {% render header.headerArea(data.piece) %}
+{% endblock %}
+
+{% block main %}
+    <!-- Post Content-->
+    <article class="mb-4">
+        <div class="container px-4 px-lg-5">
+            <div class="row gx-4 gx-lg-5 justify-content-center">
+                <div class="col-md-10 col-lg-8 col-xl-7">
+                    {% area data.piece,
+                    'main' %}
+                </div>
+            </div>
+        </div>
+    </article>
+{% endblock %}
+```
+:::
+<template v-slot:caption>
+modules/blog-page/views/show.html
+</template>
+</AposCodeBlock>
+
+As outlined above, the header of the blog piece pages is different from the other pages. There are two ways that we could approach this. We could create a dedicated blog header fragment and add it into our `views/fragments/header.html` file with a new name. The other approach is to add some conditional markup to our existing header fragment. In this case, we will do the latter.
+
+<AposCodeBlock>
+
+```django
+{% fragment headerArea(data) %}
+{% set background = apos.image.first(data.headerImage) %}
+
+{% set type = data.type %}
+
+<!-- Page Header-->
+{% if background %}
+{% set backgroundUrl = apos.attachment.url(background, { size: 'max' }) %}
+<header class="masthead" style="background-image: url('{{ backgroundUrl }}')">
+{% endif %}
+  <div class="container position-relative px-4 px-lg-5">
+    <div class="row gx-4 gx-lg-5 justify-content-center">
+      <div class="col-md-10 col-lg-8 col-xl-7">
+        <div class={{ "post-heading" if type === 'blog' else "site-heading"}}>
+          <h1>{{ data.heading }}</h1>
+          {% if type === 'blog' %}
+          <h2 class="subheading">{{ data.subheading }}</h2>
+          <span class="meta">Posted by <a href="{{ data.authorLink }}">{{ data.author }}</a> on {{ data.publicationDate }} </span>
+          {% else %}
+          <span class="subheading">{{ data.subheading }}</span>
+          {% endif %}
+        </div>
+      </div>
+    </div>
+  </div>
+</header>
+{% endfragment %}
+```
+
+<template v-slot:caption>
+views/fragments/header.html
+</template>
+</AposCodeBlock>
+
+So, what did we change? First, at the top, we created a new variable, `type`, and set it to the value of `data.type`. This is going to give us the type of page/piece that is being loaded. Looking at the code differences in the header section of the `post.html` template compared to the other three pages, the first thing we see is that the container for the header text has a class of `post-heading` instead of `site-heading`. In the code above, we conditionally set the class only if the page being loaded is of the type `blog`. Similarly, we only add the metadata if we are on this type of page.
```

Comment:
```suggestion
So, what did we change? First, at the top, we created a new variable, `type`, and set it to the value of `data.type`. This is going to give us the type of page or piece that is being loaded. Looking at the code differences in the header section of the `post.html` template compared to the other three pages, the first thing we see is that the container for the header text has a class of `post-heading` instead of `site-heading`. In the code above, we conditionally set the class only if the page being loaded is of the type `blog`. Similarly, we only add the metadata if we are on this type of page.
```

I do `thing-a/thing-b` a lot but maybe doesn't make sense here. Not sure if just changing it to `or` is clear but maybe there's a better way to word this.

Reply:
Changed.